docs/source/development/release_process.md

```diff
@@ -0,0 +1,183 @@
+---
+jupytext:
+  formats: md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.13.8
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: pyrealm_python3
+---
+
+# Package release process
+
+The package release process has three stages:
+
+* Merge changes from the `develop` branch onto `main` that will form the new release
+  version.
+* Publish a release on GitHub - this is basically just a specific tagged commit on
+  `main` that has some associated release notes.
+* Publish the built code packages to PyPI - this is the packaged version of the code
+  that users will install and use. The `pyrealm` package uses the trusted publishing
+  mechanism to make it easy to add new release to PyPI.
+
+## Generate the code commit to be released
+
+The release process for new versions of the `pyrealm` package is managed using
+pull requests to the `main` branch to create a specific commit that will be released.
+The steps of the process are:
+
+1. **Create a new release branch** from the `develop` branch called `release/X.Y.Z` ,
+   where `X.Y.Z` is the expected release version number.
+
+1. Update the `pyproject.toml` file to use the expected release versions number and
+   commit that change. You can use `poetry version` command to increment the major,
+   minor and patch version but it is almost as easy to edit the file by hand.
+
+1. That commit should set the standard `pyrealm_ci.yaml` actions going, which includes
+   code QA, testing and docs building. However, you should also check the documentation
+   builds on Read The Docs.
+
+   Log in to [https://readthedocs.org](https://readthedocs.org) which is the admin site
```

Comment:
Does this need to be the specific PyRealm readthedocs link?

Reply:
No - it's the login that takes you to the dashboard page for managing projects. There is an issue with credentials and who has access.